src/domains/setDomain.ml

```diff
@@ -366,3 +384,43 @@ struct
   let top () = of_list Elems.elems
   let is_top x = equal x (top ())
 end
+
+(** Set abstracted by a single (joined) element. *)
+module Joined (E: Lattice.S): S with type elt = E.t =
+struct
+  type elt = E.t
+  include E
+
+  let singleton e = e
+  let of_list es = List.fold_left E.join (E.bot ()) es
+  let exists p e = p e
+  let for_all p e = p e
+  let mem e e' = E.leq e e'
+  let choose e = e
+  let elements e = [e]
+  let remove e e' =
+    if E.leq e' e then
+      E.bot ()
+    else
+      e'
+  let map f e = f e
+  let fold f e a = f e a
+  let empty () = E.bot ()
+  let add e e' = E.join e e'
+  let is_empty e = E.is_bot e
+  let union e e' = E.join e e'
+  let diff e e' = remove e' e
+  let iter f e = f e
+  let cardinal e =
+    if is_empty e then
+      0
+    else
+      1
```

Comment:
This is also a bit dubious

Reply:
True, but this is at least internally consistent, e.g. `elements`/`iter`/etc all operate with this number of elements. `HoareDomain` has always been like this as well, i.e. such operations only account for the actually stored maximal elements, not all the possibly subsumed elements.

To make this behavior more explicit, I documented this both in the `SetDomain` interface and the specific implementations.